Library/Homebrew/language/java.rb

```diff
@@ -1,16 +1,23 @@
 module Language
   module Java
-    def self.java_home_cmd(version = nil)
-      version_flag = " --version #{version}" if version
-      "/usr/libexec/java_home#{version_flag}"
+    def self.java_home_cmd(_version = nil)
+      raise NotImplementedError
+    end
+
+    def self.java_home(version = nil)
```

Comment:
Do you see this being a new method that's called by formulae? If not, could move it somewhere else less public or add a `@private` at least.

Reply:
9/14 uses of `java_home_cmd` look like…
```ruby
    cmd = Language::Java.java_home_cmd("1.8")
    ENV["JAVA_HOME"] = Utils.popen_read(cmd).chomp
```
A function that immediately evaluates to `java_home` (like the Linux implementation) seems useful.
```ruby
    ENV["JAVA_HOME"] = Language::Java.java_home("1.8")
```
That doesn't however work with the current macOS implementation of `java_home`, which evaluates to the string `"$(/usr/libexec/java_home …)"`, and not the actual home.

I'll take a stab at making a useful `java_home` function and hiding the private one that possibly evaluates to `$(…)` on macOS.

Reply:
On second though, it's not worth creating a new function for just 9 uses in Homebrew/core. I'll make `java_home` private.

Reply:
I'm game for implementing it, but the current implementation is not sufficient. On macOS,
```ruby
ENV["JAVA_HOME"] = Language::Java.java_home("1.8")
```
would evaluate to 
```ruby
ENV["JAVA_HOME"] = "$(/usr/libexec/java_home --version 1.8"
```
which is incorrect. It should evaluate to
```ruby
ENV["JAVA_HOME"] = "/Library/Java/JavaVirtualMachines/jdk1.8.0_181.jdk/Contents/Home"
```

Reply:
> I'm game for it being public if it'd be used for that. I think 9 uses is sufficient 👍

`Language::Java.java_home` is now implemented, public, and tested.